contrib/olivere/elastic/elastictrace.go

```diff
@@ -48,15 +49,13 @@ func (t *httpTransport) RoundTrip(req *http.Request) (*http.Response, error) {
 	)
 	defer span.Finish()
 
-	contentLength, err := strconv.Atoi(req.Header.Get("Content-Length"))
-	if req.Body != nil && err != nil && contentLength < maxContentLength {
-		buf, err := ioutil.ReadAll(req.Body)
-		req.Body.Close()
+	if req.Body != nil {
+		buf := bufio.NewReaderSize(req.Body, maxContentLength)
+		snapshot, err := buf.Peek(maxContentLength)
 		if err != nil {
```

Comment:
Let's not return an error here. We don't want to interrupt the caller's request. Instead, only set the tag if the `err == nil`.

Reply:
I was thinking the same thing.